runtime/common/src/parachains.rs

```diff
@@ -64,23 +68,12 @@ use system::{
 use crate::attestations::{self, IncludedBlocks};
 use crate::registrar::Registrar;
 
-pub const KEY_TYPE: KeyTypeId = KeyTypeId(*b"para");
-
-pub mod crypto {
-	use super::KEY_TYPE;
-	use sp_runtime::{
-		app_crypto::{app_crypto, sr25519},
-		traits::Verify,
-	};
-	use sp_core::sr25519::Signature as Sr25519Signature;
-	app_crypto!(sr25519, KEY_TYPE);
-
-	pub struct AuthorityId;
-	impl system::offchain::AppCrypto<<Sr25519Signature as Verify>::Signer, Sr25519Signature> for AuthorityId {
-		type RuntimeAppPublic = Public;
-		type GenericSignature = Sr25519Signature;
-		type GenericPublic = sp_core::sr25519::Public;
-	}
+// An `AppCrypto` type to facilitate submitting signed transactions.
+pub struct FishermanAuthorityId;
+impl system::offchain::AppCrypto<<Signature as Verify>::Signer, Signature> for FishermanAuthorityId {
+	type RuntimeAppPublic = FishermanId;
+	type GenericSignature = sr25519::Signature;
```

Comment:
This is really over-complicated code...

Reply:
Noted here: https://github.com/paritytech/substrate/issues/5662 Unfortunately using existing traits (`IsWrappedBy`) is non trivial and it is still to be addressed in substrate. But yeah, this code is definitely redundant.

Reply:
https://github.com/paritytech/substrate/blob/999f4832e715f20a17fc2a3060bbe2ec88e7dbff/primitives/application-crypto/src/lib.rs#L204 this is even much easier.